KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -28,15 +28,32 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     }
 
     func refreshToken(completion: @escaping (Result<Bool>) -> Void) {
-        self.callback = completion
+        callback = completion
 
         guard tokenNeedsRefreshing() else {
             completion(Result.success(result: false))
             return
         }
 
-        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken else {
-            completion(Result.failure(error: RefreshTokenError.noRefreshToken))
+        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken, refreshToken.isEmpty == false else {
+            Karhoo.configuration.requestNewAuthenticationCredentials { [weak self] newCredentials in
+                guard let self = self else {
+                    completion(.failure(error: RefreshTokenError.memoryAllocationError))
+                    return
+                }
+
+                if let newCredentials = newCredentials {
+                    let newToken = AuthToken(
+                        accessToken: newCredentials.accessToken,
+                        expiresIn: Int(newCredentials.expiryDate?.timeIntervalSinceNow ?? 0),
+                        refreshToken: newCredentials.refreshToken ?? "",
+                        refreshExpiresIn: Int(newCredentials.refreshTokenExpiryDate?.timeIntervalSinceNow ?? 0)
+                    )
+                    self.handleRefreshRequest(result: .success(result: newToken))
+                } else {
+                    completion(Result.failure(error: RefreshTokenError.noRefreshToken))
```

Comment:
This error isn't descriptive of what we're after. We don't want a refresh token, we want an access token. Please rephrase it it. And sorry for missing this earlier

Reply:
Renamed to `RefreshTokenError.noAccessToken`